File: packages/python-sdk/src/blindfold/regex/detectors/ar/cuit.py
"""Argentine CUIT (Clave Unica de Identificacion Tributaria) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import ar_cuit_checksum


@register_region("ar")
class ArCuitDetector(RegexDetector):
    entity_type = EntityType.AR_CUIT
    score = 0.85
    context_keywords = [
        "cuit", "cuil", "clave unica", "identificacion tributaria",
    ]
    context_required = True
    context_window = 50

    # XX-XXXXXXXX-X format
    pattern = re.compile(
        r"\b\d{2}-?\d{8}-?\d\b"
    )
    validator = staticmethod(ar_cuit_checksum)

File: packages/js-sdk/src/regex/detectors/ar/cuit.ts
// Argentine CUIT (Clave Unica de Identificacion Tributaria) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { arCuitChecksum } from '../../validators'

class ArCuitDetector extends RegexDetector {
  entityType = EntityType.AR_CUIT
  score = 0.85
  contextKeywords = ['cuit', 'cuil', 'clave unica', 'identificacion tributaria']
  contextRequired = true

  // XX-XXXXXXXX-X format
  pattern = /\b\d{2}-?\d{8}-?\d\b/g
  validator = arCuitChecksum
}

registerRegion('ar', ArCuitDetector)

export { ArCuitDetector }
